android/app/src/main/java/com/breez/client/plugins/breez/BreezBackup.java

```diff
@@ -0,0 +1,362 @@
+
+    private final int AUTHORIZE_ACTIVITY_REQUEST_CODE = 84;
+    private final Activity m_activity;
+
+    private EventChannel.EventSink m_eventsListener;
+    private MethodChannel m_methodChannel;
+    private MethodChannel.Result m_result;
+
+    private GoogleSignInClient m_googleSignInClient;
+    private DriveClient m_driveClient;
+    private DriveResourceClient m_driveResourceClient;
+
+    public BreezBackup(PluginRegistry.Registrar registrar, Activity activity) {
+        this.m_activity = activity;
+        m_methodChannel = new MethodChannel(registrar.messenger(), BREEZ_BACKUP_CHANNEL_NAME);
+        m_methodChannel.setMethodCallHandler(this);
+
+        new EventChannel(registrar.messenger(), BREEZ_BACKUP_STREAM_NAME).setStreamHandler(this);
+
+        registrar.addActivityResultListener(this);
+
+        m_googleSignInClient = buildGoogleSignInClient();
+        getDriveClients(m_googleSignInClient.silentSignIn());
+    }
+
+    @Override
+    public void onListen(Object args, final EventChannel.EventSink events){
+        m_eventsListener = events;
+    }
+
+    @Override
+    public void onCancel(Object args) {
+        m_eventsListener = null;
+    }
+
+    private GoogleSignInClient buildGoogleSignInClient() {
+        GoogleSignInOptions signInOptions =
+            new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestScopes(Drive.SCOPE_APPFOLDER)
+                .build();
+        return GoogleSignIn.getClient(m_activity, signInOptions);
+    }
+
+    private void getDriveClients(Task<GoogleSignInAccount> task) {
+        Log.i(TAG, "Trying to execute sign in task");
+        task.addOnSuccessListener(
+            googleSignInAccount -> {
+                Log.i(TAG, "Sign in success");
+                m_driveClient = Drive.getDriveClient(m_activity, googleSignInAccount);
+                m_driveResourceClient =
+                        Drive.getDriveResourceClient(m_activity, googleSignInAccount);
+
+            })
+            .addOnFailureListener(
+                e -> {
+                    Log.w(TAG, "Sign in failed", e);
+                });
+    }
+
+    @Override
+    public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == AUTHORIZE_ACTIVITY_REQUEST_CODE) {
+            if (resultCode == RESULT_OK) {
+                getDriveClients(m_googleSignInClient.silentSignIn());
+            }
+            return true;
+        }
+        return false;
+    }
+
+    private void promptToAuthorize() {
+        Intent signInIntent = m_googleSignInClient.getSignInIntent();
+        m_activity.startActivityForResult(signInIntent, AUTHORIZE_ACTIVITY_REQUEST_CODE);
+    }
+
+    @Override
+    public void onMethodCall(MethodCall call, MethodChannel.Result result) {
+        if (call.method.equals("authorize")) {
+            if (m_driveResourceClient == null) {
+                promptToAuthorize();
```

Comment:
Here is a case where we can save the m_result because it invokes a separate activity and we can use the m_result to success/fail in the "onActivityResult" callback depends on the status so the flutter will get the right result.